src/Graph.java

```diff
@@ -0,0 +1,90 @@
+import java.util.ArrayList;
+import java.util.Map;
+
+/**
+ * {@code GraphKernel} enhanced with secondary methods.
+ */
+public interface Graph extends GraphKernel {
+    /**
+     * Returns all neighbors of a certain vertex.
+     *
+     * @param vertex
+     *            the tag of the vertex
+     *
+     * @requires {@code vertex} is a non-negative integer and Vertex with tag
+     *           {@code vertex} is in the arraylist.
+     * @return a map of the neighbors of the vertex with tag {@code vertex}.
+     */
+    Map<Integer, Integer> getNeighbors(int vertex);
+
+    /**
+     * Returns the weight of a certain edge.
+     *
+     * @param from
+     *            the tag of the vertex where the edge starts
+     * @param to
+     *            the tag of the vertex where the edge ends
+     *
+     * @requires {@code from} and {@code to} are non-negative integers. Vertices
```

Comment:
@requires is helpful but not required (I would keep it). @ensures is required, and I don't see it in any of these contracts.